### PR TITLE
Use `require_relative` in the library files

### LINKED ISSUE #2077: Prefer `require_relative`.
Not only is it more efficient, it's also more predictable.

### PR DESCRIPTION
**Pull request by marcotc**
Fixes #2077 

`require_relative` works in environments where the `$LOAD_PATH` is not set up correctly and is faster than `require`, as it doesn't have to walk the `$LOAD_PATH` list.

Running our `Gem loading` benchmarks yielded no significant change in timing or memory footprint.

This change is only required for files that are shipped in our gem package: support file and test files don't need to make this change.

The only non-trivial change are in `spec/datadog/profiling_spec.rb`, where `require` calls were asserted: I've updated them to `require_relative` assertions.

### CODE REVIEW

File: lib/datadog/tracing/contrib/http/integration.rb [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-require 'ddtrace/version'
+require_relative '../configuration/resolvers/pattern_resolver'
+require_relative 'circuit_breaker'
+require_relative 'configuration/settings'
+require_relative 'patcher'
+require_relative '../integration'
+require_relative '../../../../ddtrace/version'
 
 module Datadog
   module Tracing
```
Comment (ivoanjo) [inline lib/datadog/tracing/contrib/http/integration.rb:11]: Minor: ⬇️ The `MINIMUM_VERSION` logic here makes no sense at all. The `.version` is also incorrect, seeing as nowadays `net-http` can even be installed and updated separately from Ruby (https://rubygems.org/gems/net-http).

**Review by ivoanjo — APPROVED** [review]
👍  LGTM.

I think we should at some point remove some of the unneeded requires, as well as possibly centralizing them in the individual product-wise requires -- e.g. `datadog/appsec.rb` requires everything that appsec needs, rather than being spread out among multiple files (except from contrib) -- BUT we can do so separately from this PR.

File: lib/datadog/core.rb [hunk 2]
```diff
@@ -4,57 +4,57 @@
...
+# require_relative 'core/utils/only_once'
+# require_relative 'core/utils/sequence'
+# require_relative 'core/utils/string_table'
+# require_relative 'core/utils/time'
+# require_relative 'core/worker'
+# require_relative 'core/workers/async'
+# require_relative 'core/workers/interval_loop'
+# require_relative 'core/workers/polling'
+# require_relative 'core/workers/queue'
+# require_relative 'core/workers/runtime_metrics'
```
[the remark this answers is not in the document]
Comment (marcotc) [reply lib/datadog/core.rb:55]: I'm not sure what this list is for, but didn't want to leave it "behind". I don't have strong opinions on it.

File: lib/datadog/tracing/contrib/action_cable/event.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # typed: true
 
-require 'datadog/tracing'
-require 'datadog/tracing/context'
-require 'datadog/tracing/contrib/analytics'
-require 'datadog/tracing/contrib/active_support/notifications/event'
-require 'datadog/tracing/contrib/action_cable/ext'
+require_relative '../../../tracing'
```
[the remark this answers is not in the document]
Comment (marcotc) [reply lib/datadog/tracing/contrib/action_cable/event.rb:3]: Looks more prominent now, eh.

File: lib/datadog/tracing/contrib/http/integration.rb (hunk 1 above)
Comment (marcotc) [reply lib/datadog/tracing/contrib/http/integration.rb:11]: I'll capture this in an issue.

Comment (marcotc) [reply lib/datadog/tracing/contrib/http/integration.rb:11]: https://github.com/DataDog/dd-trace-rb/issues/2184

File: lib/datadog/appsec/contrib/auto_instrument.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # typed: true
 
-require 'ddtrace'
+require_relative '../../../ddtrace'
```
[the remark this answers is not in the document]
Comment (marcotc) [reply lib/datadog/appsec/contrib/auto_instrument.rb:3]: Done.

**Pull request merged**